src/datasets/builder.py

```diff
@@ -798,7 +799,10 @@ def _download_and_prepare(self, dl_manager, verify_infos, **prepare_split_kwargs
                     + "\nOriginal error:\n"
                     + str(e)
                 ) from None
-
+            # If check_duplicates is set to True , then except DuplicatedKeysError
+            except DuplicatedKeysError as e:
+                e.args = (e.args[0] + f"datasets/{self.name}/{self.name}.py",)
+                raise (e) from None
```

Comment:
```suggestion
            except DuplicatedKeysError as e:
                raise DuplicatedKeysError(
                    e.key,
                    e.duplicate_key_indices,
                    fix_msg=f"To avoid duplicate keys, please fix the dataset script {self.name}.py",
                ) from None
```